app/controllers/api/v4/converts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class Api::V4::ConvertsController < Api::V4::ApplicationController
+
+  before_action :check_parameters, only: [:create]
+
+  def create
+    run_file = RunFile.for_convert(params[:file])
+
+    @run = Run.new(run_file: run_file, user: nil)
+    @run.parse(fast: ["on", "1"].include?(params[:historic]) ? false : true, convert: true)
+
+    extension = {'livesplit' => 'lss',
+      'urn' => 'json',
+    }[params[:format]] || params[:format]
+
+    if params[:format] == "json"
+      new_file = render_to_string(json: @run, serializer: Api::V4::Convert::RunSerializer)
+    else
+      new_file = render_to_string("runs/#{params[:format]}", layout: false)
+    end
+    send_data(
+      new_file,
+      filename: "#{params[:file].original_filename.split(".")[0..-1].join}.#{extension}",
+      layout: false
+    )
```

Comment:
If I convert a file named `sms-any%.lss` to e.g. JSON, I get a file named `sms-any%lss.json` back, but I expect to get a file called `sms-any%.json`.